tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java

```diff
@@ -1323,6 +1323,42 @@ public static String runMain(CompileResult compileResult, String... args) {
         return exitDetails.consoleOutput;
     }
 
+    public static String runMainWithJavaOpts(CompileResult compileResult, String[] javaOpts, String... args) {
```

Comment:
Here you can change existing runMain with (CompileResult compileResult, String[] javaOpts, String... args)  and then create another method same as existing runMain and call the  the new runMAin passing empty array